features/cart/shopping_cart/adding_products_with_quantity_to_cart.feature

```diff
@@ -15,3 +15,11 @@ Feature: Adding a simple product of given quantity to the cart
         Then I should be on my cart summary page
         And I should be notified that the product has been successfully added
         And I should see "T-shirt banana" with quantity 5 in my cart
+
+    @ui
+    Scenario: Adding way too many products sets their quantity to 9999
+        Given there are 100000 units of product "T-shirt banana" available in the inventory
+        When I add 20000 of them to my cart
+        Then I should be on my cart summary page
+        And I should be notified that the product has been successfully added
```

Comment:
We should add some constraint message about it. Otherwise, it could be considered as a bug.